lib/ansible/modules/cloud/amazon/ecs_taskdefinition.py

```diff
@@ -241,10 +241,11 @@ def main():
     task_mgr = EcsTaskManager(module)
     results = dict(changed=False)
 
-    for container in module.params['containers']:
-        if 'environment' in container:
-            for environment in container['environment']:
-                environment['value'] = str(environment['value'])
+    if isinstance(module.params.get('containers'), list):
+        for container in module.params['containers']:
```

Comment:
LGTM

To simplify this more, you could remove line 244 and do:
`for container in module.params.get('containers', []):`

You could do that below as well:
`for environment in container.get('environment', []):`